sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -1157,6 +1183,33 @@ public virtual Response<ExtractLinkedEntitiesResultCollection> ExtractEntityLink
         #endregion
 
         #region Common
+
+        private static Dictionary<string, int> CreateIdToIndexMap(IEnumerable<TextDocumentInput> inputs)
+        {
+            var map = new Dictionary<string, int>();
+
+            int i = 0;
+            foreach (var item in inputs)
+            {
+                map[item.Id] = i++;
+            }
+
+            return map;
+        }
+
+        private static Dictionary<string, int> CreateIdToIndexMap(IEnumerable<DetectLanguageInput> inputs)
```

Comment:
What's the difference between this method and a previous one?

Reply:
One takes TextDocumentInput and one takes DetectLanguageInput.  I didn't want to introduce a new type or interface to the public API to unify them.  What's needed for this method is that the input has an Id field.  I'm open to suggestions on this.

Reply:
You can have an internal interface for this (methods are private anyway).

Reply:
You could use pattern matching:

```csharp
foreach (var item in inputs)
{
  string id = item switch
  {
    TextDocumentInput tdi => tdi.Id,
    DetectLanguageInput dti => dli.Id,
    _ => throw NotSupportedException(),
  };
}
```